src/NHSD.BuyingCatalogue.Ordering.Api/Services/CreatePurchasingDocument/AttachmentCsvWriter.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Threading.Tasks;
+using CsvHelper;
+using CsvHelper.Configuration;
+using CsvHelper.TypeConversion;
+
+namespace NHSD.BuyingCatalogue.Ordering.Api.Services.CreatePurchasingDocument
+{
+    public sealed class AttachmentCsvWriter<T, T2> : IAttachmentCsvWriter<T> where T2 : ClassMap<T>
+    {
+        public async Task WriteRecordsAsync(Stream stream, IEnumerable<T> records)
```

Comment:
Do we want to allow `NULL` records?

Reply:
I thought about this. It doesnt specify? I guess not, as an order wouldnt be able to be complete if there were no records. So will add a null check in